rs/src/graph/iter.rs

```diff
@@ -0,0 +1,143 @@
+use std::collections::{btree_set, BTreeSet};
+use std::{iter, slice};
+
+use super::{Edge, Graph, Subgraph, Vertex};
+
+// Iterator over the graph adjacency lists
+pub struct GraphIter<'g, 's, V, E>
+where
+    V: Vertex,
+    E: Edge,
+    's: 'g,
+{
+    pub(super) graph: &'g Graph<V, E>,
+    pub(super) iter: iter::Enumerate<slice::Iter<'g, V>>,
+    pub(super) masks: Option<(&'s Vec<bool>, &'s BTreeSet<(usize, usize)>)>,
```

Comment:
Nothing wrong with making these `pub(super)` really, but an apporach i've been using on one of my other projects would be to have these private to this file, and put the definition of `Graph::iter` in this module than the parent module.

Lets you keep the details of the iterator 100% private, which is nice.  Although guess it could make it harder to find all the functions of `Graph` so not entirely sold on it.  Not even suggesting you need to do it here, it's just an interesting thing you can do in rust that I'm not sure I've encountered in many other langs.

Reply:
I did think about that, but ultimately decided not to, partly because while I thought it's possible, I wasn't sure entirely how, and partly because having all the methods in one place seems like a slight advantage to me.